src/@next/Modal/ModalProvider.tsx

```diff
@@ -5,9 +5,19 @@ import { ModalContext, ShowModalProps } from './ModalContext';
 export const ModalProvider = ({ children }: { children: React.ReactNode }) => {
   const [state, setState] = useState<ModalProps>({ isOpen: false });
 
-  const open = (props: ShowModalProps) => setState({ ...props, isOpen: true });
+  const open = (props: ShowModalProps) => {
+    setState({ ...props, isOpen: true });
 
-  const close = () => setState({ isOpen: false });
+    if (typeof window != 'undefined' && window.document) {
```

Comment:
do we need this change? Wouldn't Modal component will handle this?

Reply:
oops, you're right. we don't need these